tst/plistqueue.tst

```diff
@@ -0,0 +1,79 @@
+gap> q := PlistQueue();
+<queue with 0/64 entries>
+
+#
+gap> PushFront(q, 15);
+gap> PopFront(q);
+15
+gap> PushBack(q, 15);
+gap> PopBack(q);
+15
+gap> Push(q, 15);
+gap> Pop(q);
+15
+gap> IsEmpty(q);
+true
+
+# test size, make sure it is bigger than
+# initial capacity so that expansion happens
+gap> N:=67;;
+
+# add at the front, pop at the back
+gap> for i in [1..N] do PushFront(q,i); od;
+gap> IsEmpty(q);
+false
+gap> Size(q) = N;
+true
+gap> out := List([1..N], x -> PopBack(q));;
+gap> out = [1..N];
+true
+gap> IsEmpty(q);
+true
+
+# add at the front, pop at the front
+gap> for i in [1..N] do PushFront(q,i); od;
+gap> IsEmpty(q);
+false
+gap> Size(q) = N;
+true
+gap> out := List([1..N], x -> PopFront(q));;
+gap> out = [N,N-1..1];
+true
+gap> IsEmpty(q);
+true
+
+# add at the back, pop at the front
+gap> for i in [1..N] do PushBack(q,i); od;
+gap> IsEmpty(q);
+false
+gap> Size(q) = N;
+true
+gap> out := List([1..N], x -> PopFront(q));;
+gap> out = [1..N];
+true
+gap> IsEmpty(q);
+true
+
+# add at the back, pop at the back
+gap> for i in [1..N] do PushBack(q,i); od;
+gap> IsEmpty(q);
+false
+gap> Size(q) = N;
+true
+gap> out := List([1..N], x -> PopBack(q));;
+gap> out = [N,N-1..1];
+true
+gap> IsEmpty(q);
+true
+
+
```

Comment:
Perhaps insert something like this here: `q := PlistQueue(32);` That would serve multiple purposes: It would test this alternative constructor; it would exercise the code path in `PushBack` which leads to queue expansion (which currently is not covered, because by the time we start doing some serious `PushBack`, the queue has long grown to a size were it can all fit in).